README.md

```diff
@@ -1,30 +1,90 @@
-# constellation
-A blockchain microservice operating system
-
-# Building for development
-#### Instructions tested on Mac and Linux (may work elsewhere)
-1. Check out repository (git clone git@github.com:Constellation-Labs/constellation.git)
-2. From constellation directory, run ./build.sh -- optionally ./build.sh seedhost:port (to connect to other host)
-3. Interact with app via the api (by default http://localhost:9000, look at API.scala for endpoints)
-
-# Docker execution (assuming you have docker already set up)
-##### Install docker locally -- for now not covering that here.
-##### We will publish docker images to dockerhub soon(ish).
+# Constellation
+A blockchain microservice operating system.
+
+This is the reference implementation supporting our 
+[DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph) 
+protocol, writting in the 
+[Scala](https://www.scala-lang.org/) 
+programming language.
+
+```
+constellation
+├── scripts                                <─── shell scripts
+└── src
+    ├── it                                 <─── continuous integration
+    ├── main/scala/org/constellation
+        ├── consensus
+        ├── crypto
+        ├── API.scala
+        └── LevelDB.scala
+    └── test                               <─── unit tests
+├── build.sh
+└── run-local-docker.sh
+```
+
+## :computer: Building for development
+### On Mac and Linux
+1. Check out repository 
+```haskell
+git clone git@github.com:Constellation-Labs/constellation.git
+```
+2. From root directory ``constellation, run 
+```haskell
```

Comment:
I don't think you need Haskell in here? just the back ticks

Reply:
True, no language annotation required. It should be shell or bash but they don't have shell and when I didn't find the language list for md markup, I didn't want to make 5 commits to test. Haskell is my default syntax highlighting goto when I want to emphasise /.({ and numbers from text and that's why I went for thus substitude. Can be removed or maybe someone knows which shell varient they recognise.

Reply:
ah gotcha, makes sense